infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -311,14 +313,17 @@ private void applySnapshotSync(LogReplicationEntry message) {
         switch (message.getMetadata().getMessageMetadataType()) {
             case SNAPSHOT_MESSAGE:
                 snapshotWriter.apply(message);
-                return;
+                break;
             case SNAPSHOT_END:
+                dataConsistent.set(0);
+                logReplicationMetadataManager.setDataConsistentOnStandby(String.valueOf(dataConsistent));
```

Comment:
I am a little confused about why the dataConsistent is initialized as 1.
If a SNAPSHOT_MESSAGE comes, dataConsistent is still 1. Is it correct?

Reply:
Yes, dataConsistent is 1 in any phase other than apply phase of the snapshot sync.  This is because only during apply phase we have an intermediate state where replicated data is being applied and applications can read inconsistent data.